### DISCUSSION TITLE
PHP Warning:  Version warning: Imagick was compiled against Image Magick version 1650 but version 1654 is loaded. Imagick will run but may behave surprisingly in Unknown on line 0

### DISCUSSION THREAD

**Discussion by Kushalitha** [ImageMagick/ImageMagick#2022] [Help]
I recently updated to php 7.3. now I'm getting above error code. could anyone tell me how to fix this. My Imagick in as below. Thanks

imagick module version | 3.4.3
imagick classes | Imagick, ImagickDraw, ImagickPixel, ImagickPixelIterator
Imagick compiled with ImageMagick version | ImageMagick 6.7.2-7 2017-03-22 Q16 
Imagick using ImageMagick library version | ImageMagick 6.7.6-8 2016-07-19 Q16

**Comment by fmw42** [comment]
Imagick is trying to use a different version of ImageMagick than you have on your system. Sorry, I know little about Imagick.  Do you have ImageMagick 6 or 7 on your system?  Was Imagick installed for ImageMagick 6 or 7. Perhaps that is the conflict. Otherwise it is just a minor version difference.

**Reply by Kushalitha** [reply]
Thanks for the reply

**Reply by Kushalitha** [reply]
I have added the Imagick version info to my previous thread. Please have a look

**Reply by fmw42** [reply]
I am not an expert on Imagick. But these lines 

> Imagick compiled with ImageMagick version | ImageMagick 6.7.2-7 2017-03-22 Q16
Imagick using ImageMagick library version | ImageMagick 6.7.6-8 2016-07-19 Q16

Imply a conflict of ImageMagick versions.  Imagick was compile to use  6.7.2-7, but you only have 6.7.6-8 on your system, a version that is much newer.

I suspect you need to install the same version of Imagick on your systems as was Imagick compiled against.


Do your scripts work or not?

**Reply by Kushalitha** [reply]
I contact host but the advisor has no idea. then he reverse my php version back to 7.1 all website didn't work. took whole day to fix the website. finally fixed the webisite and upgraded to php 7.3 but now I get this error. but don't no how to fix

**Reply by fmw42** [reply]
I am sorry, but I do not know anything more to suggest. To me it looks like you have to back down your imagemagick on your system to 6.7.2-7. Or install a newer version of Imagick compatible with 6.7.6-8 that you have on your system.

**Comment by clockworkroger** [comment]
I know this is years old now, but for what it's worth, my solution (MacOS Homebrew install) was reinstalling the pecl package
```shell
pecl uninstall imagick; pecl install imagick
``` 

Then it complained a bit about having imagick loading multiple times, so I just removed the duplicate entries in `/usr/local/etc/php/*/php.ini`

Hope that helps anyone who lands here like I did.

**Reply by jtomek** [reply]
This solved the issue. Thanks

**Reply by JanMalte** [reply]
If you have multiple PHP versions installed with `homebrew` you might need to run the command with every version specific `pecl` command:
```
/opt/homebrew/opt/php@7.4/bin/pecl uninstall imagick
/opt/homebrew/opt/php@7.4/bin/pecl install imagick

/opt/homebrew/opt/php@8.0/bin/pecl uninstall imagick
/opt/homebrew/opt/php@8.0/bin/pecl install imagick
```

**Reply by tkivelip** [reply]
Thanks guys. Was helpful :)

**Reply by fmahadyBD** [reply]
but what will do a window user ?

**Comment by Kushalitha** [comment]
Since I'm using shared hosting, I don't have direct access to my php.ini or imagick.ini files. I contacted the host, but the advisor didn't know how to help. So, what I did was create a **.user.ini** file and add the code `imagick.skip_version_check=true`. I then saved it in the public_html directory. That solution worked for me.

**Comment by JuniorTak** [comment]
I recently encountered a similar PHP Warning. This is due to a mismatch between the version of ImageMagick installed and the version that Imagick compiles with.

To fix it, uninstall ImageMagick from your programs then check the ImageMagick version that Imagick compiles with in the **Imagick** section of **phpinfo()**, line **Imagick compiled with ImageMagick version**. Download and install that version. That's it!

You can find the different versions of ImageMagick here https://www.npackd.org/p/imagemagick64

I hope this helps :)

**Reply by fmahadyBD** [reply]
Imagick was compiled against ImageMagick version 1808 but version 1809 is loaded. Imagick will run but may behave surprisingly in Unknown on line 0

what is this means?
i am try different ways but i can't
now php version is  8.2.12 in xampp. Please

**Reply by iamdevhnoo** [reply]
Imagick was compiled against ImageMagick version 1808 but version 1809 is loaded. Imagick will run but may behave surprisingly in Unknown on line 0
---
Has anyone solved this?
And how to solve it?

**Reply by fmw42** [reply]
There is no such versions of Imagemagick. Perhaps that is the versions of Imagick. Obviously the wrong version of Imagemagick is being used.  How did you install Imagick and how did you install Imagemagick. I believe that usually one just installs Imagick and it install Imagemagick that is compatible. Perhaps you already had a different version of Imagemagick on your system before installing Imagick. Check that you only have one version of Imagemagick. Perhaps remove both and start again with Imagick install.

**Reply by Kushalitha** [reply]
> ## Imagick was compiled against ImageMagick version 1808 but version 1809 is loaded. Imagick will run but may behave surprisingly in Unknown on line 0
> 
> Has anyone solved this? And how to solve it?

When the Imagick extension is installed, it’s compiled against a specific version of ImageMagick. In your issue, it was compiled against version 1808. However, when PHP runs, it detects that the current installed version of ImageMagick is 1809, which could cause issues due to incompatibilities between the two versions. Reinstall or update Imagick so that it is compiled against the correct version of ImageMagick (1809).

**Reply by puapeap** [reply]
> When the Imagick extension is installed, it’s compiled against a specific version of ImageMagick. In your issue, it was compiled against version 1808. However, when PHP runs, it detects that the current installed version of ImageMagick is 1809, which could cause issues due to incompatibilities between the two versions. Reinstall or update Imagick so that it is compiled against the correct version of ImageMagick (1809).

Hi @fmw42,
 
this solved my issue (pictures could not be uploaded in Piwigo, even though ImageMagick was correctly shown as installed under "Enviroment"). The only hint was a warning in the php_error_log of XAMPP  "PHP Warning:  Version warning: Imagick was compiled against ImageMagick version 1808 but version 1809 is loaded. Imagick will run but may behave surprisingly in Unknown on line 0"

after replacing the php_imagick.dll file in the ext folder of php and restarting XAMPP it worked.

**Reply by Pixel-Patch** [reply]
> > When the Imagick extension is installed, it’s compiled against a specific version of ImageMagick. In your issue, it was compiled against version 1808. However, when PHP runs, it detects that the current installed version of ImageMagick is 1809, which could cause issues due to incompatibilities between the two versions. Reinstall or update Imagick so that it is compiled against the correct version of ImageMagick (1809).
> 
> Hi @fmw42,
> 
> this solved my issue (pictures could not be uploaded in Piwigo, even though ImageMagick was correctly shown as installed under "Enviroment"). The only hint was a warning in the php_error_log of XAMPP "PHP Warning: Version warning: Imagick was compiled against ImageMagick version 1808 but version 1809 is loaded. Imagick will run but may behave surprisingly in Unknown on line 0"
> 
> after replacing the php_imagick.dll file in the ext folder of php and restarting XAMPP it worked.

Hello, I have the same error what version did you download and replace the php_imagick.dll file 

I currently have: 
ImageMagick-7.1.1-Q16-HDRI and php_imagick-3.7.0-8.2-ts-vs16-x64

**Comment by pakceramic** [comment]
my website is also show this error
PHP Warning: Version warning: Imagick was compiled against ImageMagick version 1692 but version 1693 is loaded
https://pakcontractor.com/

**Reply by fmw42** [reply]
This is an Imagemagick forum, not one for iMagick.  Please report on an iMagick forum
